Robust_Lab.qmd

```diff
@@ -128,10 +341,69 @@ df <- data.frame(hours=c(1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4,
 
 a\. Use the lm() function to fit a regression model in R that uses **hours** as the predictor variable and **score** as the response variable
 
+```{r}
+mod = lm(score ~ hours, data = df)
+
+mod %>% 
+  broom::tidy() %>%
+  gt()
+```
+
 b\. Interpret the results
 
-c\. Check assumptions and report which ones failed (include plots)
+**The results of the model suggest that exam scores increase by 1.95 for each additional hour spent studying. However, the increase is not significant (p = 0.09).**
+
+**Marginal effect**
+
+c\. Check assumptions and report which assumptions are violated (include stats or plots)
+
+## (Residual) Linearity and Homoscedasticity
+
+```{r}
+#1 - Residual linearity and homoscedasticity
+performance::check_model(mod, check = c("linearity", "homogeneity"))
+```
+
+**The plots above suggest that the assumptions of (residual) linearity and homoscedasticity are not met by the data. The residuals and the predicted values do not fall along a straight line. In other words, the model has larger errors at different predicted values. The violation of the linearity assumption can also be read from a scatter plot of the *raw* values of the DV and IV:**
+
+```{r}
+df %>% ggplot(aes(hours, score)) +
+  geom_point(size = 4) +
+  geom_smooth(method = "loess", se = FALSE, color = '#84a59d') +
+  ggtitle("Scatter Plot of Exam Scores and Hours Spent Studying") +
+  ylab("Exam Score") +
+  xlab("Hours")
+```
+
+## Normality of Residuals
+
+```{r}
+#Normality of residuals
+performance::check_model(mod, check = c("qq", "normality"))
+```
+
+**These plots suggest that a second assumption, the normality of residuals, is also being violated by our current model. The Q-Q plot suggests that the majority of points do not fall along the straight line and that, as a result, the model is making poorer predictions for those values. Additionally, the density plot does not appear to follow a normal distribution (although a Shaprio-Wilk test returns a p \> .05).**
+
+## Outliers
+
+```{r}
+performance::check_model(mod, check = "outliers")
+```
+
+**While an absence of outliers is not an assumption of linear models *per se*, it is good practice to examine the data for their presence. The plot above suggests that one influential case may be causing trouble for our model (red point, outside lower dashed contour).**
+
+**As we are dealing with only one predictor, we need not be concerned with the no multicolinearity assumption and (assuming the data is between-subjects) the independence of observations assumption.**
 
 d\. Re-run the lm you saved above, but with robust standard errors
 
+```{r}
+mod_robust = coeftest(mod, vcov = vcovHC(mod, type = "HC0"))
```

Comment:
HC0 vs. HC3?

Reply:
I went for HC0 because I (maybe mistakenly?) thought it was the standard convention. Love to hear your thoughts on pros and cons of each?

Reply:
I have commonly seen HC3 being used. What I have seen is that for smaller sample sizes HC3 is a bit better.

Reply:
Ah okay, handy to know. Thanks so much!